compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1592,14 +1592,14 @@ impl<'tcx> Place<'tcx> {
         self.projection.iter().any(|elem| elem.is_indirect())
     }
 
-    /// If MirPhase >= Derefered and if projection contains Deref,
-    /// It's guaranteed to be in the first place
-    pub fn has_deref(&self) -> bool {
+    /// If MirPass >= Derefered and if projection contains Deref,
+    /// Deref must to be in the first place.
+    ///
+    /// This function checks the statement above, and must only be called
```

Comment:
The comment is wrong or confusing. It says that the function checks that "if there is a deref it is in the first place", but that's not what the function checks.

Reply:
Simplified the wording I think it's clearer now.